src/langchain_google_spanner/document_loader.py

```diff
@@ -0,0 +1,348 @@
+                    content_columns,
+                    metadata_columns,
+                    metadata_json_column,
+                    row,
+                )
+
+
+class SpannerDocumentSaver:
+    """Save docs to Google Cloud Spanner."""
+
+    def __init__(
+        self,
+        instance: str,
+        database: str,
+        table_name: str,
+        client: Client = Client(),
+        content_column: Optional[str] = "",
+        metadata_columns: Optional[List[str]] = [],
+        metadata_json_column: Optional[str] = "",
+    ):
+        """Initialize Spanner document saver.
+
+        Args:
+            instance: The Spanner instance to load data to.
+            database: The Spanner database to load data to.
+            table_name: The table name to load data to.
+            client: The connection object to use. This can be used to customized project id and credentials.
+            content_column: Optional. The name of the content column. Defaulted to the first column.
+            metadata_columns: Optional. This is for user to opt-in a selection of columns to use. Defaulted to use
+                              all columns.
+            store_metadata: If true, extra metadata will be stored in the "langchain_metadata" column.
+            metadata_json_column: Optional. The name of the special JSON column. Defaulted to use "langchain_metadata".
+        """
+        self.instance = instance
+        self.database = database
+        self.table_name = table_name
+        self.content_column = content_column
+        self.metadata_columns = metadata_columns
+        self.metadata_json_column = metadata_json_column or METADATA_COL_NAME
+        self.client = client
+        spanner_instance = self.client.instance(instance)
+        if not spanner_instance.exists():
+            raise Exception("Instance doesn't exist.")
+        spanner_database = spanner_instance.database(database)
+        if not spanner_database.exists():
+            raise Exception("Database doesn't exist.")
+        spanner_database.reload()
+        self.dialect = spanner_database.database_dialect
+        spanner_table = spanner_database.table(table_name)
+        if not spanner_table.exists():
+            raise Exception(
+                "Table doesn't exist. Create table with SpannerDocumentSaver.init_document_table function."
+            )
+        self._table_fields = [
+            n.name for n in spanner_table.schema if n.name != self.metadata_json_column
+        ]
+        self._table_fields.append(self.metadata_json_column)
+
+    def add_documents(self, documents: List[Document]):
+        """Add documents to the Spanner table."""
+        db = self.client.instance(self.instance).database(self.database)
+        values = [
+            _load_doc_to_row(self._table_fields, doc, self.metadata_json_column)
+            for doc in documents
+        ]
+
+        for values_batch in _batch(values, MUTATION_BATCH_SIZE):
```

Comment:
For this method we want to stay under 5 MB of data. I think the MUTATION_BATCH_SIZE should work

Reply:
Sounds good, right now I have it set for 1000 rows.